src/media/local-stream.spec.ts

```diff
@@ -0,0 +1,108 @@
+import { BaseEffect } from '@webex/web-media-effects';
+import MediaStreamStub from '../mocks/media-stream-stub';
+import MediaStreamTrackStub from '../mocks/media-stream-track-stub';
+import { mocked } from '../mocks/mock';
+import { createMockedStream } from '../util/test-utils';
+import { LocalStream } from './local-stream';
+
+/**
+ * Gain access to private variables using typescript's `any`.
+ *
+ * @param item - The item to gain access to.
+ * @returns The value entered.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const privateAccess = (item: any): any => item;
+
+/**
+ * Create a mocked track effect.
+ *
+ * @returns A mocked TrackEvent.
+ */
+const createMockedTrackEffect = () => {
+  const effectTrack = mocked(new MediaStreamTrackStub());
+  const effect = {
+    dispose: jest.fn().mockResolvedValue(undefined),
+    load: jest.fn().mockResolvedValue(effectTrack),
+    on: jest.fn(),
+  };
+
+  return { effectTrack, effect };
+};
+
+/**
+ * A dummy LocalStream implementation so we can instantiate it for testing.
+ */
+class TestLocalStream extends LocalStream {}
+
+describe('LocalStream', () => {
+  const mockStream = createMockedStream();
+  let localStream: LocalStream;
+  beforeEach(() => {
+    localStream = new TestLocalStream(mockStream);
+  });
+
+  it('should change the underlying track state based on being muted & unmuted', () => {
+    expect.assertions(2);
+    // Simulate the default state of the track's enabled state.
+    mockStream.getTracks()[0].enabled = true;
+
+    localStream.setMuted(true);
+    expect(privateAccess(localStream).outputStream.getTracks()[0].enabled).toBe(false);
```

Comment:
can't we verify these on `mockStream` instead of using `privateAccess`?

Reply:
Updated.

Reply:
i think this change may have gotten lost?

Reply:
I changed the above code to:

```ts
mockStream.getTracks()[0].enabled = true;

localStream.setMuted(true);
expect(mockStream.getTracks()[0].enabled).toBe(false);
localStream.setMuted(false);
expect(mockStream.getTracks()[0].enabled).toBe(true);
```